homeassistant/components/google_cloud/tts.py

```diff
@@ -153,3 +196,65 @@ async def async_get_tts_audio(self, message, language, options):
             extension = "wav"
 
         return extension, response.audio_content
+
+
+class GoogleCloudTTSEntity(BaseGoogleCloudProvider, TextToSpeechEntity):
+    """The Google Cloud TTS entity."""
+
+    def __init__(
+        self,
+        entry: ConfigEntry,
+        client: texttospeech.TextToSpeechAsyncClient,
+        voices: dict[str, list[str]],
+        language: str,
+        options_schema: vol.Schema,
+    ) -> None:
+        """Init Google Cloud TTS entity."""
+        super().__init__(client, voices, language, options_schema)
+        self._attr_unique_id = f"{entry.entry_id}-tts"
+        self._attr_name = entry.title
+        self._attr_device_info = dr.DeviceInfo(
+            identifiers={(DOMAIN, entry.entry_id)},
+            name=entry.title,
+            manufacturer="Google",
+            model="Cloud",
+            entry_type=dr.DeviceEntryType.SERVICE,
+        )
+        self._entry = entry
+
+    async def async_get_tts_audio(
+        self, message: str, language: str, options: dict[str, Any]
+    ) -> TtsAudioType:
+        """Load TTS from Google Cloud."""
+        try:
+            return await self._async_get_tts_audio(message, language, options)
+        except GoogleAPIError as err:
+            _LOGGER.error("Error occurred during Google Cloud TTS call: %s", err)
+            if isinstance(err, Unauthenticated):
+                self._entry.async_start_reauth(self.hass)
+            return None, None
+
+
+class GoogleCloudTTSProvider(BaseGoogleCloudProvider, Provider):
+    """The Google Cloud TTS API provider."""
+
+    def __init__(
+        self,
+        client: texttospeech.TextToSpeechAsyncClient,
+        voices: dict[str, list[str]],
+        language: str,
+        options_schema: vol.Schema,
+    ) -> None:
+        """Init Google Cloud TTS service."""
+        super().__init__(client, voices, language, options_schema)
+        self.name = "Google Cloud TTS"
```

Comment:
Is this an entity?

Reply:
Yes

Reply:
Then we should not set `self.name` directly

Reply:
Wait, are you sure? I thought Provider is not an entity.

Reply:
Sorry this is the legacy provider. The one above is the entity.